tools/new-component/index.js

```diff
@@ -55,38 +55,42 @@ module.exports = class extends Generator {
       // To access props later use this.props.someAnswer;
       props.dashlessName = props.name.replace(/-/g, '');
       props.underscoreName = props.name.replace(/-/g, '_');
-      props.camelCaseName = _.camelCase(props.name);
+      props.camelCaseName = camelCase(props.name);
       props.cleanPatternType = props.patternType.replace(/([0-9])\w+-/g, '');
       this.props = props;
     }.bind(this));
     /* eslint-enable func-names, prefer-arrow-callback, no-param-reassign */
   }
 
   writing() {
+    const {
+      files, patternType, patternSubType, name,
+    } = this.props;
+
     const destPath = path.join(
       patternBase,
-      this.props.patternType,
-      this.props.patternSubType,
-      this.props.name,
+      patternType,
+      patternSubType,
+      name,
     );
 
-    if (_.includes(this.props.files, 'scss')) {
+    if (files.includes('scss')) {
```

Comment:
Instead of these repeated functions that are effectively doing the exact same thing, refactor so we simply have an object config like:

```js
const generatorAssets = {
  twig: {
    templatePath: '_pattern.twig',
    destinationPath: path.join(destPath, `_${name}.twig`),
  },
  // ... same for scss, js, etc
}
```

Use this as a lookup on a loop.

Reply:
Completed in 5bd809d71811fe7b4d27d3a9c66a85436d1723b7